packages/react-components/theme-designer/src/components/ColorTokens/ColorTokensList.tsx

```diff
@@ -80,12 +84,18 @@ export const ColorTokensList: React.FunctionComponent<ColorTokensListProps> = pr
 
         const handleColorChange: MenuProps['onCheckedValueChange'] = (e, data) => {
           const newColor = parseInt(data.checkedItems[0] as string, 10) as Brands;
-          dispatchColorOverrides({ colorToken: color, newValue: newColor });
+          dispatchState({ type: 'Override', overrides: { [color]: brand[newColor] } });
```

Comment:
Consider removing `dispatchState` and `dispatchColorOverride` from `ColorTokensList`'s props in favor on an `onChange` event handler prop. Both of these dispatch functions are for controlling application state and this component doesn't need to know about them. Rather it should do the conversion for `newColor` and then call `onChange` so the parent component can handle updating state. Something like (pseudo code below):

```jsx
// ColorToken.tsx

const onColorChange = (....) => {
    dispatchState();
    dispatchColorOverride();
};

....

<ColorTokensList
   // other props
   onChange={onColorChange}/>
```

```jsx
// ColorTokensList.tsx

const handleColorChange = (....) => {
    const newColor = parseInt(data.checkedItems[0] as string, 10) as Brands;
    onChange?.(newColor, color, ...anything else we need to pass ...);
};
```